src/main/kotlin/com/codingchallenge/transactionanalyser/TransactionLoader.kt

```diff
@@ -0,0 +1,35 @@
+package com.codingchallenge.transactionanalyser
+
+import com.codingchallenge.transactionanalyser.domain.Transaction
+import com.codingchallenge.transactionanalyser.domain.TransactionBuilder
+import java.io.BufferedReader
+import java.io.InputStream
+import java.io.InputStreamReader
+import java.util.stream.Collectors
+
+class TransactionLoader(private val input: InputStream) {
+    fun loadTransactionsFromCsv(): List<Transaction>{
+        val transactions: List<Transaction>
+        BufferedReader(InputStreamReader(input)).use {
+            transactions = it.lines()
+                .skip(1)
+                .map(this::buildTransactionFromCsvLine)
+                .collect(Collectors.toList())
+        }
+        return transactions
+    }
+
+    private fun buildTransactionFromCsvLine(line: String): Transaction{
```

Comment:
is there a better way we could have thought about passing the values from the CSV through? 
Here we are using magic numbers https://stackoverflow.com/questions/47882/what-is-a-magic-number-and-why-is-it-bad 
Be careful with this, and think about how the data might scale, if new columns are added then we would have to change every number after that column change.
Could we perhaps have used the kotlin-csv library and mapped the headings of the CSV itself, and used those names to map to the builder here rather than with 0,1,2,3 etc? or any other way?

Reply:
Yeah, magic numbers are bad. I didn't use external CSV libraries for the simplicity sake. I think there are libraries out there that can handle this nicely (i.e. annotation on the class which maps fields to csv columns). Will have a look at what kotlin-csv offers.